tests/5.2/declare_mapper/test_declare_mapper_present.c

```diff
@@ -0,0 +1,67 @@
+//===---- test_declare_mapper_present.c -------------------------------------===//
+// 
+// OpenMP API Version 5.2 
+//
+// This example has been adapted from the 5.2 OpenMP Examples document:
+// "declare mapper Directive", and "OpenMP Directive Syntax -> complex iterator"
+// The declare mapper directive will be used to automatically map variables
+// according to its prescription:  full structure, plus the dynamic storage of the
+// data element. These variable should be present on the device.
+//
+//===-------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1000
+
+typedef struct myvec{
+    size_t len;
+    double *data;
+} myvec_t;
+
+
+#pragma omp declare mapper(default:myvec_t v) map(present, tofrom: v, v.data[0:v.len]) 
```

Comment:
I would put this to replace line 44 under my target enter data suggesstion. Thus, the mapper is declared after the struct has already been mapped to the device, otherwise the map would fail as it would inherently require itself to be present. Change the tofrom to from as the the struct has already been mapped to the device.